javalib/src/main/scala/java/util/Arrays.scala

```diff
@@ -617,11 +616,11 @@ object Arrays {
   }
 
   @inline private def checkIndicesForCopyOfRange(
-      len: Int, start: Int, end: Int): Unit = {
+      a: Array[_], start: Int, end: Int): Unit = {
```

Comment:
You should `@specialize` this function if you do that, otherwise it will go through a run-time type test to do `a(start)`.